src/state/wallets/walletsStore.ts

```diff
@@ -0,0 +1,664 @@
+              logger.warn('[walletsStore]: declaring selected wallet unhealthy...');
+              set({
+                selected: wallets[wallet.id],
+              });
+            }
+            logger.debug('[walletsStore]: done updating wallets');
+          }
+        }
+
+        if (!healthyKeychain) {
+          captureMessage('Keychain Integrity is not OK');
+        }
+
+        logger.debug('[walletsStore]: check completed');
+        saveKeychainIntegrityState('done');
+      } catch (e) {
+        logger.error(new RainbowError("[walletsStore]: error thrown'"), {
+          message: (e as Error)?.message,
+        });
+        captureMessage('Error running keychain integrity checks');
+      }
+    },
+
+    getWalletWithAccount(accountAddress: string): RainbowWallet | undefined {
+      const { wallets } = get();
+      if (!wallets) {
+        return;
+      }
+      const sortedKeys = Object.keys(wallets).sort();
+      let walletWithAccount: RainbowWallet | undefined;
+      const lowerCaseAccountAddress = accountAddress.toLowerCase();
+      sortedKeys.forEach(key => {
+        const wallet = wallets[key];
+        const found = wallet.addresses?.find((account: RainbowAccount) => account.address?.toLowerCase() === lowerCaseAccountAddress);
+        if (found) {
+          walletWithAccount = wallet;
+        }
+      });
+      return walletWithAccount;
+    },
+
+    getAccountProfileInfo(props) {
+      const { selected, walletNames, getWalletWithAccount } = get();
+      const wallet = props?.wallet || (props ? getWalletWithAccount(props.address) : null) || selected;
+      const accountAddress = props?.address || get().accountAddress;
+
+      if (!wallet || !accountAddress || !wallet?.addresses?.length) {
+        return {
+          accountAddress,
+          accountColor: 0,
+        };
+      }
+
+      const accountENS = walletNames?.[accountAddress];
+      const lowerCaseAccountAddress = accountAddress.toLowerCase();
+      const selectedAccount = wallet.addresses?.find(
+        (account: RainbowAccount) => account.address?.toLowerCase() === lowerCaseAccountAddress
+      );
+
+      if (!selectedAccount) {
+        return {
+          accountAddress,
+          accountColor: 0,
+        };
+      }
+
+      const { label, color, image } = selectedAccount;
+      const labelWithoutEmoji = label && removeFirstEmojiFromString(label);
+      const accountName = labelWithoutEmoji || accountENS || address(accountAddress, 4, 4);
+      const emojiAvatar = returnStringFirstEmoji(label);
+      const accountSymbol = returnStringFirstEmoji(emojiAvatar || addressHashedEmoji(accountAddress));
+      const accountColor = color;
+      const accountImage = image && isValidImagePath(image) ? image : null;
+
+      return {
+        accountAddress,
+        accountColor,
+        accountENS,
+        accountImage,
+        accountName,
+        accountSymbol,
+      };
+    },
+  }),
+  {
+    storageKey: 'walletsStore',
```

Comment:
underneath we're keeping the keychain as the source of truth, do we still want this persist store?

Reply:
yea i remembered wanting to delete this before i left and forgot to push